pkg/agent/core/ngt/service/ngt.go

```diff
@@ -450,10 +452,12 @@ func (n *ngt) insert(uuid string, vec []float32, t int64, validation bool) (err
 		err = errors.ErrUUIDNotFound(0)
 		return err
 	}
-	if validation && !n.vq.DVExists(uuid) {
-		// if delete schedule exists we can insert new vector
-		_, ok := n.kvs.Get(uuid)
-		if ok || n.vq.IVExists(uuid) {
+	if validation {
+		ovec, err := n.GetObject(uuid)
+		if err == nil {
+			if f32stos(vec) == f32stos(ovec) {
+				return nil
+			}
```

Comment:
the insert time will change, so I think it is better not to add this logic

Reply:
I am sorry if I made a mistake. kpango added validation logic of `DVExits`  to `GetObject`. so I think this implementation is okay. What do you think?

Reply:
I thought it would be no problem since I already checked DeleteQueue by doing GetVector as well, but I think I'll turn it off this time since it's effective enough even if I check only when updating.